packages/react-compose/src/resolveSlotProps.ts

```diff
@@ -25,12 +29,19 @@ export function resolveSlotProps<TProps, TState>(
     });
   });
 
-  //  Mix unrecognized props onto root, appropriate, excluding the handled props.
-  assignToMapObject(
-    slotProps,
-    'root',
-    getNativeElementProps(slots.root, state, [...(options.handledProps as string[]), 'className']),
-  );
+  //  Mix unrecognized state onto root, excluding the handled props.
+  Object.keys(state).reduce<Partial<TState>>((acc, propName: string) => {
+    if (
+      !defaultHandledProps.has(propName) &&
+      !options.slots.hasOwnProperty(propName) &&
+      propName !== 'as' &&
+      (options.handledProps as string[]).indexOf(propName) === -1
```

Comment:
this could be changed into a set lookup potentially

Reply:
This should reduce bundle size and certainly perf overhead but will put a strong reliance on handledProps.